lib/pure/asyncdispatch.nim

```diff
@@ -136,10 +156,51 @@
 ## Examples
 ## ========
 ##
-## For examples take a look at the documentation for the modules implementing
+## .. code-block:: nim
+##   proc example() {.async.} =
+##     echo "asyncCheck Example"
+##
+##   asyncCheck example()
+##
+## For more examples take a look at the documentation for the modules implementing
 ## asynchronous IO. A good place to start is the
 ## `asyncnet module <asyncnet.html>`_.
 ##
+##
+## waitFor
+## -------
+##
+## ``waitFor`` is similiar to ``asyncCheck`` but it will wait for the ``Future``.
+##
+## The usage and features are the same of ``asyncCheck`` (mentioned above). Example:
+##
+## .. code-block:: nim
+##   proc example() {.async.} =
+##     echo "waitFor Example"
+##
+##   waitFor example()
+##
+## await
+## -----
+##
+## ``await`` will wait for the ``Future``, and gives you the contents inside the ``Future``,
+## ``await`` only available inside a function with the ``{.async.}`` or ``{.multisync.}`` Pragmas.
+##
+## The usage and features are the same of ``waitFor`` (mentioned above).
```

Comment:
usage is the same, I don't understand what features there are to speak of which are the same.